### DISCUSSION TITLE
Bugs and feature

### DISCUSSION THREAD

**Discussion by akaKarthick** [czbar/ChessForge#807] [General]
1. I request to add the “Go to Bottom/ Back to Top” floating button In “Chapter” view.
2. Option to change font size in the search result of “Find Identical Position”. Currently it's too small.
3. And show the “bookmark ribbon/ flag/star” icon somewhere near a board, if the position is bookmarked. So it will cut down the time of right click and check the position is already bookmarked by checking “remove bookmark”.  Or at least colourful underline the notation of that position.
4. Currently, In the “Games” view, a particular heading area, User can switch the games by clicking left and right arrows. I will request you to include the mouse scroll activity in that particular area for a faster way to switch games. It will make it easy to Jump 10 games rather than clicking 10 times or going back to chapter and scroll down/up to switch other games which are not close to the current game I am exploring. 
5. And I found one more issue to rectify. I opened the “Find Position” tool. And I manually(from my memory) input the position I want to search. After that i select “search”. Expected behaviour is Show the result, if no result were found it should be back to the position i inputted. Because I can change a bit and try again. But currently it closes the “Find Position”. So i have to reopen and once again i have to input the position(from mind) i want to search. So it's annoying to reopen and retry. So i req you make software to keep the position i previously imputed to change. Give up or Retry is user's choice.
6. Lately, I like to play Chess 960 more than the standard. So I am using other softwares like “Nibbler”. But I can't control the difficulty in that software. Lichess gives Stockfish 1-8 levels so I can't control the difficulty. So please add the Chess 960 as well.
7. One more issue to rectify. Select any of the games and choose the winning side (position) to play, select “Start training from here”. And deliver a checkmate. Then Evaluate the line you played. It will evaluate wrongly if there is a force checkmate. Wrongly In the sense checkmate for -# or +#. Save the game and check the scoresheet. I'll add the video later if you can't repro. And dont turn on the engine. “Create exercise from there”. And “Guess the next move” or “Enter the analysis and submit” you can find more bugs.
8. And also add a timer feature for “Play against engine”. Because it will help time management.
9. And I don't know the name of this thing. I Guess “NAG” . annotation thing in bottom.  In the exercise view, I entered the “Guess the next move” option. And playing moves. Then I tried that annotation thing. I can use it for engine moves. But i cant use it for my moves. Because the engine moves fastly. Even i changed the moving time to 10 sec i cant input annotation. I guess there is a moving time issue.
10. And also “Promote line” and a few others are freezed by default in the exercise view, but then you check the solution then hide it. Now the “Promote line” and other things are unfreezed. Btw, not effective. just telling
11. And if you select the piece instead of solving mode, a popup window is showing. With yes or no options. I cant close that window by closing btn. Expected behaviour, it should close whereever i click other than clicking yes btn.

choose yourself which is bug to fix or feature to add.

Ignore my poor english. and I'll add more observation later.

**Comment by czbar** [comment]
I have been super busy with non-chess related things recently but hope to resume Chess Forge development in a few days.  In the meantime, just a quick response to the items you have raised.

Re 1:**FIXED** Nice idea. By "floating", do you mean positioned somewhere directly over the view?  Well, it should be possible although I'll need to check for some possible technical challenges.

Re 2: **FIXED** Ok, I'll either increase the font or give an option for the user to increase it. (**FIX**: increased the default font to the standard size, in line with the text in the Views).

Re 3: This is easy to implement ... I just need to find the right place and the right icon!

Re 4: **FIXED** Actually, I find the inability to quickly navigate through multiple games very annoying too :)   However, there is a deeper underlying issue. After each arrow press, the program attempts to render the next game and it can be slow (mostly due to the slowness of the Microsoft's Rich Text Box control that I cannot do much about).  Multiple presses means multiple renderings and the program appears almost frozen for a second or so.  Allowing mouse scroll function won't help that, I'm afraid.  My plan here is to change the behavior so that when you click on the "next/previous" arrow but do not release the mouse button, the program will rapidly increase/decrease the game number and eventually render the game at which you released the button.  This is a common behavior in many GUI's so it will be familiar to the users.  (**WORK STARTED**: per my preferred solution as described above).

Re 5: **FIXED** Ok, I'll keep the edited position if nothing was found.  But if something was found, then the next search (SHIFT-F3) will reset to the current position, otherwise it would be confusing to the user.  Alternatively, I can offer a drop-down list of previously searched positions, perhaps?  (_As a workaround, you can copy FEN before search and then paste it in the search dialog later on._)  (**FIX**:   if nothing found the user will be prompted to edit the search, if something found the "Found Positions" dialog will have a button to "Search again" i.e. go back to position editing.)

Re 6: Yeah, although I personally enjoy standard chess much more (still have not mastered it :)), I do understand the recent spike in the interest in 960.  Implementation-wise it is no big deal (just setting the starting position and adjusting castling rules, I believe). I would need to figure out how to blend it into the current functionality without confusing the users.  I'll see.  For now, with lower priority.

Re 7: **CANNOT REPRO or DON'T UNDERSTAND** Ok, I'll take a look when I'm back working on the program.

Re 8: Not too sure about this one.  Does not seem aligned with the current, limited purpose of the program. But if one day Chess Forge is to expand its training scope (e.g. adding features currently well out of scope like, say, "spaced repetition" and such), then that may be a feature to be added there too.

Re 9, 10: **FIXED** I'll take a look.  (**FIX**: The "NAG bar" has no use when in Solving Node so I am simply hiding it. The menus enable/diable status has been fixed but I am sure there some inconsistencies there somewhere still.)

Re 11: **FIXED** Yes, this is an oversight, I'll fix it.

**Comment by akaKarthick** [comment]
Currently, is there any notification at bottom when the user copy/paste line? i am unable to access windows machines for few more moths.

**Comment by czbar** [comment]
I just added a "flash notification" a couple of weeks ago so it's not in the released version yet. Will be in the next one.
 
![image](https://github.com/czbar/ChessForge/assets/7270924/b4e2f6a7-c53f-43ed-9064-0d562c79a6a5)

==

_Btw I'll release the next version once I'm done with the new Study layout_

![image](https://github.com/czbar/ChessForge/assets/7270924/5e7ef76c-e560-4887-928c-5ee1a438ecf0)

**Comment by akaKarthick** [comment]
and add "Undo" flash notification for for deletion actions. Just like this
[
![Screenshot from 2024-03-06 12-27-12](https://github.com/czbar/ChessForge/assets/125844120/5083df78-4efe-49f6-b1c0-ce157ed13ea9)
](url)
